src/runtime/internal/style_manager.ts

```diff
@@ -1,9 +1,11 @@
 import { element } from './dom';
 import { raf } from './environment';
 
-let stylesheet;
+type DocStyles = [CSSStyleSheet, Record<string, true>];
+
+let activeDocs = new Set<Document>();
+let docStyles = new WeakMap<Document, DocStyles>();
```

Comment:
We're not using `WeakMap` anywhere else currently, and I think we probably want that to continue being the case since it's basically impossible to polyfill in environments that don't support it. Reckon we could get away with this being a `Map` instead? In the vast majority of cases we're only going to have a single key, and it's always likely to be stable, so *maybe* it doesn't matter if we hang on to references?